tools/fastp/fastp.xml

```diff
@@ -0,0 +1,207 @@
+        <param name="qualified_quality_phred" type="integer" optional="true" label="Qualified quality phred" help="The quality value that a base is qualified. Default 15 means phred quality >=Q15 is qualified."/>
+        <param name="unqualified_percent_limit" type="integer" optional="true" label="Unqualified percent limit" help="How many percents of bases are allowed to be unqualified (0~100). Default 40 means 40%."/>
+        <param name="n_base_limit" type="integer" optional="true" label="N base limit" help="If one read's number of N base is >n_base_limit, then this read/pair is discarded. Default is 5."/>
+        <param name="disable_length_filtering" type="boolean" truevalue="-L" falsevalue="" checked="false" label="Disable length filtering" help="Length filtering is enabled by default. If this option is specified, length filtering is disabled."/>
+        <param name="length_required" type="integer" optional="true" label="Length required" help="Reads shorter than this value will be discarded, default is 15."/>
+        <param name="correction" type="boolean" truevalue="-c" falsevalue="" checked="false" label="Enable base correction" help="Enable base correction in overlapped regions (only for PE data), default is disabled."/>
+        <param name="umi" type="boolean" truevalue="-U" falsevalue="" checked="false" label="Enable unique molecular identifer" help="Enable unique molecular identifer (UMI) preprocessing."/>
+        <param name="umi_loc" type="text" optional="true" label="UMI location" help="Specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none."/>
+        <param name="umi_len" type="integer" optional="true" label="UMI length" help="If the UMI is in read1/read2, its length should be provided."/>
+        <param name="umi_prefix" type="text" optional="true" label="UMI prefix" help="If specified, an underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG, final=UMI_AATTCG). No prefix by default."/>
+        <param name="overrepresentation_analysis" type="boolean" truevalue="-p" falsevalue="" checked="false" label="Enable overrepresented analysis" help="Enable overrepresented sequence analysis."/>
+        <param name="overrepresentation_sampling" type="integer" optional="true" label="Overrepresentation sampling" help="One in (--overrepresentation_sampling) reads will be computed for overrepresentation analysis (1~10000), smaller is slower, default is 20."/>
+    </inputs>
+
+    <outputs>
+        <data format="fastq" name="out1" label="${tool.name} output"/>
+        <data format="fastq" name="out2" label="${tool.name} paired end output">
+            <filter>paired_end_data == True</filter>
+        </data>
+    </outputs>
+
+    <tests>
+        <test> 
+            <param name="in1" value="R1.fq"/>
```

Comment:
Could you please add a few more tests and tests more params?

Reply:
I will, but I still have problems with this one. I don't think I read the file correctly into the tool. When I use `planemo serve`, I get the following:
```
Detecting adapter for single end input...
No adapter detected

Read1 before filtering:
total reads: 0
total bases: 0
Q20 bases: 0(-nan%)
Q30 bases: 0(-nan%)

Read1 after filtering:
total reads: 0
total bases: 0
Q20 bases: 0(-nan%)
Q30 bases: 0(-nan%)
```
when it should be like this:
```
Detecting adapter for single end input...
No adapter detected

Read1 before filtering:
total reads: 2
total bases: 302
Q20 bases: 262(86.755%)
Q30 bases: 241(79.8013%)

Read1 after filtering:
total reads: 2
total bases: 302
Q20 bases: 262(86.755%)
Q30 bases: 241(79.8013%)

Filtering result:
reads passed filter: 2
reads failed due to low quality: 0
reads failed due to too many N: 0
reads failed due to too short: 0
reads with adapter trimmed: 0
bases trimmed due to adapters: 0

JSON report: fastp.json
HTML report: fastp.html

fastp -i R1.fq -o out.fq 
fastp v0.12.3, time used: 0 seconds
```
Which indicates that the tools is reading an empty file. That is after I added single quotes to the input and text params.

Reply:
Also in the travis log, same as my `planemo test` output, it says that the difference between the expected files is that it should be missing the the lines in `out.fq`

Reply:
This is the executed command: `fastp -i /tmp/tmpR7ybQV/files/000/dataset_1.dat -o /tmp/tmpR7ybQV/files/000/dataset_2.dat --thread 4 `
Please check if this tools need to have a valid file ending.

Reply:
Yeah, I think it does:
https://github.com/OpenGene/fastp/blob/master/src/fastqreader.cpp

Will create a link file to handle it, thanks